src/edgeit.jl

```diff
@@ -0,0 +1,66 @@
+import Base: start, next, done, length, show,
+            in, ==
+
+type EdgeItState
+    it::Int
+    v::Int
+    k::Int
+end
+
+type EdgeIt
+    m::Int
+    adj::Vector{Vector{Int}}
+    start::EdgeItState # =[it, v, k]
+    directed::Bool
+end
+
+edge_it(g::Graph) = EdgeIt(ne(g), g.fadjlist, EdgeItState(1,1,1), false)
+edge_it(g::DiGraph) = EdgeIt(ne(g), g.fadjlist, EdgeItState(1,1,1), true)
+
+start(eit::EdgeIt) = eit.start
+done(eit::EdgeIt, state::EdgeItState) = state.it > eit.m
+function next(eit::EdgeIt, state::EdgeItState)
+    # println("$eit, $state")
+    assert(state.v <= length(eit.adj))
+    u = 0
+    while state.k <= length(eit.adj[state.v])
```

Comment:
Could you add some comments describing these while loops and why breaking if directed for our future selves?